source/extensions/filters/http/grpc_http1_reverse_bridge/filter.cc

```diff
@@ -136,17 +136,11 @@ Http::FilterHeadersStatus Filter::encodeHeaders(Http::ResponseHeaderMap& headers
     // If the response from upstream does not have the correct content-type,
     // perform an early return with a useful error message in grpc-message.
     if (content_type != upstream_content_type_) {
-      headers.setGrpcMessage(badContentTypeMessage(headers));
-      headers.setGrpcStatus(Envoy::Grpc::Status::WellKnownGrpcStatus::Unknown);
-      headers.setStatus(enumToInt(Http::Code::OK));
-
-      if (!content_type.empty()) {
-        headers.setContentType(content_type_);
-      }
+      decoder_callbacks_->sendLocalReply(Http::Code::OK, badContentTypeMessage(headers), nullptr,
+                                         Grpc::Status::WellKnownGrpcStatus::Unknown,
+                                         RcDetails::get().GrpcBridgeFailedContentType);
```

Comment:
I *think* this ends up setting content-type of the response to text unless the http connection manager's local_reply_config is configured with a mapper that matches and sets content-type.

Assuming that's correct, I think that's worth mentioning (or linking to docs) in the release notes.

Reply:
I think this would piggy back on the default local reply handling for gRPC and hit this code: https://github.com/envoyproxy/envoy/blob/master/source/common/http/utility.cc#L476-L492

Reply:
Ah. Was looking for that but didn't find it.

Reply:
A little late to this but do we have a integration test that covers this case? I'm slightly concerned about the sendLocalReply path having very little testing when called in the encoding path. cc @alyssawilk @snowp

Reply:
I thought we did but now that I look into it I see that there isn't, and while adding one it seems like we're hitting some ASSERT. I'll see if I can come up with a quick fix, if not I'll revert this PR.